### PR TITLE
Support Filtering Entries On List Fields

### PR DESCRIPTION
**Pull request by jbutz**
**- Summary**

When setting up a collection for a specific group of files on my site I discovered the `filter` property for collections did not work for fields that are lists. This change will allow filtering on lists when the filter value is one of the values in the list.

**- Test plan**

I modified the `dev-test/index.html` and `dev-test/config.yml` files to test this change. My modified files can be found here: https://gist.github.com/jbutz/79a6d2161102187b1fce59f47dd61d58

I added a new collection titled `FAQ - FILTERED` that has a filter value based on a new field. I added the new field to two of the FAQ documents in `index.html` and I gave one of those documents the value that will match the filter. Below is a screenshot of the result.

test-plan-screenshot

**- Description for the changelog**

Add support for filtering collections on list fields

**- A picture of a cute animal (not mandatory but encouraged)**

Say hello to Eddy

animal

### CODE REVIEW

**Comment by verythorough** [comment]
Deploy preview for *netlify-cms-www* ready!

Built with commit 8d043c4961533ddf0d7dd5189453bebd0c5553f1

https://deploy-preview-1578--netlify-cms-www.netlify.com

**Comment by verythorough** [comment]
Deploy preview for *cms-demo* ready!

Built with commit b788dc2ffb569eef733e5b9c2f8605e9ae3d332b

https://deploy-preview-1578--cms-demo.netlify.com

File: packages/netlify-cms-core/src/__tests__/backend.spec.js
Comment (erquhart) [inline packages/netlify-cms-core/src/__tests__/backend.spec.js]: First of all, **thank you** for adding tests!

I'm thinking we should add an extra entry to each of these `entries` arrays so that it's possible for more than one value to be returned and actual filtering takes place in the tests.

Comment (jbutz) [reply packages/netlify-cms-core/src/__tests__/backend.spec.js]: Good point, I'll add that as soon as I get a chance

**Comment by jbutz** [comment]
Ok @erquhart, I have added additional entries to the tests that won't watch the filters

**Comment by erquhart** [comment]
Awesome, thanks!

**Comment by erquhart** [comment]
Was hoping to get another review on this (we _try_ to require to maintainer reviews), but I'll go ahead and merge 👍 

Thanks again!

**Review by Hell081 — APPROVED** [review]

**Review by Hell081 — APPROVED** [review]
L oh

**Pull request merged**
